File: frontend/de/unika/ipd/grgen/ir/exprevals/PiExpr.java
/*
 * GrGen: graph rewrite generator tool -- release GrGen.NET 4.5
 * Copyright (C) 2003-2019 Universitaet Karlsruhe, Institut fuer Programmstrukturen und Datenorganisation, LS Goos; and free programmers
 * licensed under LGPL v3 (see LICENSE.txt included in the packaging of this file)
 * www.grgen.net
 */

package de.unika.ipd.grgen.ir.exprevals;

public class PiExpr extends Expression {
	public PiExpr() {
		super("pi expr", DoubleType.getType());
	}

	public void collectNeededEntities(NeededEntities needs) {
	}
}

File: frontend/de/unika/ipd/grgen/ir/expr/numeric/PiExpr.cs
/*
 * GrGen: graph rewrite generator tool -- release GrGen.NET 8.1
 * Copyright (C) 2003-2026 Universitaet Karlsruhe, Institut fuer Programmstrukturen und Datenorganisation, LS Goos; and free programmers
 * licensed under LGPL v3, some components/parts use different licenses (see LICENSE.txt included in the packaging of this file)
 * www.grgen.de / www.grgen.net
 */

namespace de.unika.ipd.grgen.ir.expr.numeric
{
using BuiltinFunctionInvocationExpr = de.unika.ipd.grgen.ir.expr.invocation.BuiltinFunctionInvocationExpr;
using DoubleType = de.unika.ipd.grgen.ir.type.basic.DoubleType;

public class PiExpr : BuiltinFunctionInvocationExpr
{
	public PiExpr()
		: base("pi expr", DoubleType.Type)
	{
	}
}

}
